core/src/main/java/org/elasticsearch/index/mapper/core/TypeParsers.java

```diff
@@ -284,6 +286,15 @@ public static void parseField(FieldMapper.Builder builder, String name, Map<Stri
                 builder.fieldDataSettings(settings);
                 iterator.remove();
             } else if (propName.equals("copy_to")) {
+                if (parserContext.isWithinMultiField()) {
+                    if (indexVersionCreated.after(Version.V_2_1_0) ||
+                        (indexVersionCreated.after(Version.V_2_0_1) && indexVersionCreated.before(Version.V_2_1_0))) {
+                        throw new MapperParsingException("copy_to in multi fields is not allowed. Found the copy_to in field [" + name + "] which is within a multi field.");
+                    } else {
+                        ESLoggerFactory.getLogger("mapping [" + parserContext.type() + "]").warn("Found a copy_to in field [" + name + "] which is within a multi field. This feature has been removed and the copy_to will be ignored.");
+                        // we still parse this, otherwise the message will only appear once and the copy_to removed. After that it will appear again. Better to have it always.
+                    }
+                }
                 parseCopyFields(propNode, builder);
```

Comment:
The mappings will get merged on master so if the master is an old node and still has the copy_to and the new mapping without the copy_to comes in it will be merged with the copy_to from master and the resulting mapping will still have it. Only when a new node becomes master it will actually remove the copy_to. 
My concern is actually not so much that the log message appears too often but more that it appears in any case at least once but then when a user sees it and checks the mapping then the copy_to might or might not be there depending on if the master was upgraded already or not.